### PR TITLE
Refactor mediator integration tests

### PR DESCRIPTION
**Pull request by dilee**
## Purpose
Refactoring and improving integration tests for mediators.

### CODE REVIEW

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/SoapHeaderBlocksTestCase.java [hunk 1]
```diff
@@ -159,4 +142,23 @@ public boolean verify(String hostname, SSLSession session) {
         }
         return null;
     }
+
+    private Callable<Boolean> isCarFileDeployed(final String carFileName) {
```
Comment (prabushi) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/SoapHeaderBlocksTestCase.java]: Add a method description.

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/conditionalRouter/ConditionalRouterIntegrationTest.java [hunk 2]
```diff
@@ -318,6 +313,15 @@ public void conditionalRouterMediatorWithEqualHeaderTest() throws Exception {
 
     }
 
+    private Callable<Boolean> dynamicSequenceExists (final String[] seqArr, final String strToFind) {
```
Comment (prabushi) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/conditionalRouter/ConditionalRouterIntegrationTest.java]: Add a method description.

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/conditionalRouter/ConditionalRouterIntegrationTest.java [hunk 3]
```diff
@@ -223,13 +221,10 @@ public void conditionalRouterMediatorWithDynamicSequenceTest() throws Exception
...
-                if (!ArrayUtils
-                        .contains(sequenceAdminServiceClient.getDynamicSequences(), "conf:/filters/dynamic_seq1")) {
-                    break;
-                }
-                Thread.sleep(500);
-            }
+
+            Awaitility.await().pollInterval(500, TimeUnit.MILLISECONDS).atMost(
+                    60, TimeUnit.SECONDS).until(dynamicSequenceExists(sequenceAdminServiceClient.
+                    getDynamicSequences(), "conf:/filters/dynamic_seq1"));
```
Comment (prabushi) [inline integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/conditionalRouter/ConditionalRouterIntegrationTest.java]: Define the configuration path as a constant.

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/SoapHeaderBlocksTestCase.java (hunk 1 above)
Comment (dilee) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/aggregate/SoapHeaderBlocksTestCase.java]: Fixed in https://github.com/wso2/product-ei/pull/3665

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/conditionalRouter/ConditionalRouterIntegrationTest.java (hunk 2 above)
Comment (dilee) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/conditionalRouter/ConditionalRouterIntegrationTest.java]: Fixed in https://github.com/wso2/product-ei/pull/3665

File: integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/conditionalRouter/ConditionalRouterIntegrationTest.java (hunk 3 above)
Comment (dilee) [reply integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/conditionalRouter/ConditionalRouterIntegrationTest.java]: Fixed in https://github.com/wso2/product-ei/pull/3665

**Pull request merged**